webapp/player-project/src/ws-proxy.ts

```diff
@@ -38,6 +38,17 @@ const WebSocketProxy = new Proxy(window.WebSocket, {
         }
         return Reflect.set(target, prop, value);
       },
+      get(target, prop, receiver) {
+        const value = Reflect.get(target, prop, receiver);
+        // Because these methods are part of the native WebSocket prototype,       
+        // they must be called with the original WebSocket as `this`.
+        // If they're called with the Proxy as `this`, it results in an "illegal invocation".
+        // Binding them to the underlying `target` (the real WebSocket) avoids this issue.
```

Comment:
```suggestion
        // Because these methods are part of the native WebSocket prototype,
        // they must be called with the original WebSocket as `this`.
        // If they're called with the Proxy as `this`, it results in an "illegal invocation".
        // Binding them to the underlying `target` (the real WebSocket) avoids this issue.
```